Implement Brackets popup

anatoly-os opened the pull request:
TODO:
- Crash happens on my machine when incrementing value for Column or Span property

anatoly-os commented on inspectors/models/abstractinspectormodel.cpp:
```diff
@@ -264,6 +269,10 @@ QVariant AbstractInspectorModel::valueFromElementUnits(const Ms::Pid& pid, const
         return strList.join(",");
     }
 
+    case Ms::P_TYPE::INT: {
```
Not sure whether it is needed. That was an attempt to fix the crash in slots processing.

vpereverzev replied:
yes, this is not necessary, for simple cases like QVariant <int> or QVariant <QString> enough of what is in default

anatoly-os commented on inspectors/view/qml/notation/brackets/BracketSettings.qml:
```diff
@@ -0,0 +1,37 @@
+
+    icon: {
+        var icon = IconNameTypes.BRACKET
+        if (model && model.bracketType === BracketTypes.BRACKET_TYPE_BRACE)
```
It doesn't work, unfortunately :(

vpereverzev replied:
it should be `model.bracketType.value`. You are comparing the "PropertItem" object with int

anatoly-os reviewed (commented):
The problem is that two things don't work:

1. Crash happens when I change the values in the Bracket properties
<img width="1680" alt="Screenshot 2020-05-13 at 17 42 17" src="https://user-images.githubusercontent.com/18065034/81834496-48804480-9541-11ea-970b-0ae3a4d5bdc6.png">
<img width="1680" alt="Screenshot 2020-05-13 at 17 42 15" src="https://user-images.githubusercontent.com/18065034/81834511-4e762580-9541-11ea-946a-fa740e81a5ed.png">
<img width="1680" alt="Screenshot 2020-05-13 at 17 42 11" src="https://user-images.githubusercontent.com/18065034/81834513-4fa75280-9541-11ea-9d6d-8de1bbc718ce.png">

vpereverzev commented on inspectors/view/qml/notation/brackets/BracketSettings.qml:
```diff
@@ -0,0 +1,37 @@
+
+    text: {
+        var text = qsTr("Bracket")
+        if (model && (model.bracketType === BracketTypes.BRACKET_TYPE_BRACE))
```
the same thing here, use .value of "PropertyItem" object

anatoly-os commented:
@vpereverzev when I change selection while the Inspector is opened, `isBracketTypeEqual()` method is not called until I drop the selection. It leads to not changing the popup icon and text.

I tend to reimplement the brackets and braces as separate components to avoid all these "bugs".

anatoly-os commented:
Ok, Bracket and Brace are now separate components and popups.

The crash still happens :(

vpereverzev commented on inspectors/view/qml/notation/brackets/internal/BracketPopupSection.qml:
```diff
@@ -0,0 +1,37 @@
+Column {
+
+    property QtObject intBracketProperty: undefined
+    property string labelProperty: undefined
```
it's better to initialize string by empty string 
But actually you can just use an alias property
`property alias title: textLabel.text`

anatoly-os committed:
Implemented Brackets and Braces popups

TODO:
- Crash happens on my machine when incrementing value for Column or Span property

anatoly-os commented:
The crash happen if and only if one bracket is selected when we change the column or span property. If there is multiple selection of several brackets or a bracket with different elements, the crash won't happen.
